src/transform/src/join_implementation.rs

```diff
@@ -640,6 +764,25 @@ mod differential {
                 input_mapper,
             )?;
 
+            // Count new arrangements.
+            //
+            // We collect the count for each input, to be used to calculate `new_arrangements` below.
+            let new_input_arrangements: Vec<usize> = orders
+                .iter()
+                .map(|o| {
+                    o.iter()
+                        .skip(1)
```

Comment:
I think the first input in an order actually does need an arrangement in case of a Differential join. See the `implement_arrangements` call, and then `order.remove(0);`.

(For Delta, there is a `skip(1)` _before_ `implement_arrangements`.)

Reply:
Ah, gotcha---yes, fixed.